docs/layers/eks/design-decisions/decide-on-default-storage-class.mdx

```diff
@@ -0,0 +1,63 @@
+- **Managed Service:** Fully managed by AWS, reducing operational overhead for maintenance and scaling.
+- **Availability Zone Failover**: For workloads that require failover across multiple Availability Zones, EFS is a more suitable option. It provides multi-AZ durability, ensuring that Pods can recover and access persistent storage seamlessly across different AZs.
+
+### Cons:
+- **Lower Performance:** Generally offers lower performance compared to EBS, with throughput as low as 100 MB/s, which may not meet the demands of even modestly demanding applications.
+- **Higher Cost:** Significantly more expensive than EBS, at least 3x the price per GB and potentially more depending on performance demands, although there may be some savings from not having to reserve size for future growth.
+- **Higher Latency:** Higher latency compared to EBS, which may impact performance-sensitive applications.
+- **No Native Backup Support:** EFS lacks a built-in, straightforward backup and recovery solution for EKS. Kubernetes-native tools don’t support EFS backups directly, requiring the use of alternatives like AWS Backup. Recovery, however, can be non-trivial and may involve complex configurations to restore data effectively.
+
+## Amazon EBS
+
+**Amazon EBS** provides high-performance block storage volumes for use with Amazon EC2 instances, suitable for a wide range of workloads.
+
+### Pros:
+- **Higher Performance:** Offers high IOPS and low latency, making it ideal for performance-critical applications.
+- **Cost-Effective:** Potentially lower costs for specific storage types and usage scenarios.
+- **Native EKS Integration:** Well-integrated with Kubernetes through the EBS CSI (Container Storage Interface) driver, facilitating seamless provisioning and management.
+- **Supports Snapshot and Backup:** Supports snapshotting for data backup, recovery, and cloning.
+
+### Cons:
+- **Single-Attach Limitation:** EBS volumes may only be attached to a single node at a time, limiting shared access across multiple pods.
+- **Data Sharing:** Data cannot be easily shared across multiple instances, requiring additional configurations or solutions for shared access.
+- **Availability Zones:** EBS volumes are confined to a single Availability Zone, which may affect high availability and disaster recovery strategies. Kubernetes has implemented several mitigations for this limitation, and ECS likely has similar solutions.
+- **Scalability:** Requires manual provisioning and management of storage capacity, which can introduce operational complexity.
+
+## Recommendation
+
+Use **Amazon EBS** unless there is a good reason not to.
+
+Use **Amazon EFS** when:
+
+- Multiple Pods need read/write access to the same data at the same time.
+- Persistent data needs to be available in multiple Availability Zones and the underlying application does not support replication.
```

Comment:
```suggestion
Use **Amazon EBS** as the primary storage option when:

	•	High performance, low-latency storage is required for workloads confined to a single Availability Zone.
	•	The workload doesn’t require shared access across multiple Pods.
	•	You need cost-effective storage with support for snapshots and backups.
	•	Manual resizing of volumes is acceptable for capacity management, recognizing that failover across AZs requires manual intervention and provisioning.

Consider **Amazon EFS** when:

	•	Multiple Pods need concurrent read/write access to shared data across nodes.
	•	Workloads must persist data across multiple Availability Zones for high availability, and the application does not support native replication.
	•	Elastic, automatically scaling storage is necessary to avoid manual provisioning, especially for workloads with unpredictable growth.
	•	You are willing to trade off higher costs and lower performance for multi-AZ durability and easier management of shared storage.

> ![IMPORTANT]
> EFS should never be used as backend storage for performance-sensitive applications like databases, due to its high latency and poor performance under heavy load.
```